using/self_service.md

```diff
@@ -322,3 +322,118 @@ Once allocation has been created, you can see its access information:
+4. Upload requirements.txt for installation in a virtual environment. Again, only exact versions of packages. New libraries will appear in the form, save to apply the changes.
+5. If any libraries were installed manually using pip through SSH (or via Jupyter notebooks), you need to synchronize that virtual environment with ETAIS using this button (otherwise new libraries will not appear in a virtual environmnet)
+6. If any virtual environments were manually installed/removed in Virtual environments root directory, you need to synchronize the state with ETAIS with this button (otherwise new virtual environment will not appear).
+7. A brief guide which tells you how to activate virtual environment in terminal, install libraries and that you should click “Find missing installed libraries” button if new libraries were installed manually.
+8. Each action performed by ETAIS on a VM is logged here.
+9. Expand virtual environment (shows list of libraries)
+
+## JupyterHub management service
+
+Before using JupyterHub, please setup [Python management service](#python-management-service) on a VM.
+
+Once provisioned, JupyterHub configuration file is located in ```/etc/jupyterhub/jupyterhub_config.py```. All changes manually applied to JupyterHub configuraiton files will not be reflected in JupyterHub details form. Moreover, each time you click "Apply configuration" button, the existing configuration file is completely replaced with the new one generated out of the currently saved state in ETAIS. However, the backup for old configuration file is created in the same folder.
+
+JupyterHub management service is not notified if you change, for example, admin rights through JupyterHub GUI - do it through self-service.
+
+### Details screen - manual JupyterHub user entry
+
+![JupyterHub management details Linux PAM authentication section](../images/jupyterhub-management-linux-pam.png)
+
+1. Virtual environments come from Python management service.
+2. It is possible to make these virtual environments accessible as Jupyter kernels. All JupyterHub users will have access to them. However, they will not be allowed to install or remove libraries through notebook or SSH in these kernels.
+3. There are 2 types of authentication methods available. In both cases JupyterHub users correspond to Linux system users.
+ * Linux PAM: you need to manually define JupyterHub users. Corresponding system users will be created once Apply configuration button is pressed.
+ * OAuth: see the next section. Corresponding system users will be created on first login.
+4. Apply configuration (currently active Jupyter users will experience short connection outage with JupyterHub).
+5. Passwords for already created users are not stored in ETAIS database. Password should be defined only for new users (form validation will guide you).
+
+### Details screen - OAuth2 authentication method
+
+![JupyterHub management details OAuth2 authentication section](../images/jupyterhub-management-oauth.png)
```

Comment:
Why Jupyter kernels section is empty? Why isn't there any buttons, indicators or anything else? Is it broken?

Reply:
there are just no virutal envs...